dipy/reconst/sfm.py

```diff
@@ -33,7 +33,64 @@
     warnings.warn(w)
 
 
-def sfm_design_matrix(gtab, sphere, response, mode='signal'):
+# Isotropic signal models: these are models of the part of the signal that
+# changes with b-value, but does not change with direction. This collection is
+# extensible, by inheriting from IsotropicModel/IsotropicFit below:
+class IsotropicModel(ReconstModel):
+    """
+    A base-class for the representation of isotropic signals.
+    The default behavior, suitable for single b-value data is to calculate the
+    mean in each voxel as an estimate of the signal that does not depend on
+    direction.
+    """
+    def __init__(self, gtab):
+        ReconstModel.__init__(self, gtab)
+
+    def fit(self, data):
+        data_no_b0 = data[..., ~self.gtab.b0s_mask]
+        params = np.mean(data_no_b0, -1)
+        if len(data_no_b0.shape) == 1:
+            n_vox = 1
+        else:
+            n_vox = data_no_b0.shape[0]
```

Comment:
Perhaps slightly clearer to first call

```
data_no_b0 = np.array(data_no_b0, ndmin=2, copy=False)
```

in the beginning of the function, and then this if/else falls away.